### PR TITLE
Feature: Add tickets table to event details page

### PR DESCRIPTION
**Pull request by pauloiankoski**
Resolves [GIVE-362]

## Description

This PR adds the Tickets table to the Event Details Page.

### Todo:
- Add create ticket modal
- Add edit ticket modal
- Add delete ticket endpoint

## Visuals

CleanShot 2024-03-01 at 21 28 32
CleanShot 2024-03-01 at 21 31 32

## Pre-review Checklist

-   [ ] Acceptance criteria satisfied and marked in related issue
-   [x] Relevant `@unreleased` tags included in DocBlocks
-   [ ] Includes unit tests
-   [ ] Reviewed by the designer (if follows a design)
-   [x] [Self Review](https://give.gitbook.io/development-manual/devops/github/code-reviews#self-review) of code and UX completed

[GIVE-362]: https://stellarwp.atlassian.net/browse/GIVE-362?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: src/EventTickets/resources/admin/components/EventDetailsPage/InnerPageListTable/index.tsx [hunk 1]
```diff
@@ -0,0 +1,125 @@
...
+                sortDirection: direction,
+            };
+        });
+    };
+
+    return (
+        <>
+            <ShowConfirmModalContext.Provider value={showConfirmActionModal}>
+                <div className={styles.innerPageListTable}>
+                <ListTable
```
Comment (kjohnson) [inline src/EventTickets/resources/admin/components/EventDetailsPage/InnerPageListTable/index.tsx:69]: Do we need the full list table component? Ticket Types will most likely be a single page, so no pagination, probably doesn't need column sorting or column filtering by search.

Can we just localize a list of ticket types to per-populate a table? Adding a new event can re-load the page.

Comment (pauloiankoski) [reply src/EventTickets/resources/admin/components/EventDetailsPage/InnerPageListTable/index.tsx]: Refactored. Also removed some components that had been introduced here but not used and that will be heavily changed in next PRs like: `InnerPageListTable`, `EventSectionRowActions` and `TicketTypesRowActions`.

File: src/EventTickets/resources/admin/components/EventDetailsPage/TicketTypesSection/index.tsx [hunk 2]
```diff
@@ -1,13 +1,65 @@
...
+    const imagePath = `${window.GiveEventTicketsDetails.pluginUrl}/assets/dist/images/list-table/blank-slate-event-tickets-icon.svg`;
+    return (
+        <div className={styles.container}>
+            <img src={imagePath} alt={__('No ticket created yet', 'give')} />
+            <h3>{__('No ticket created yet', 'give')}</h3>
+            <p className={styles.helpMessage}>{__('Create a ticket to complete your event setup.', 'give')}</p>
+            <p>
+                {/*Todo: Set an onClick event to open the CreateTicketTypeModal*/}
+                <a
+                    href={`${window.GiveEventTicketsDetails.adminUrl}edit.php?post_type=give_forms&page=give-event-tickets&new=event`}
```
Comment (kjohnson) [inline src/EventTickets/resources/admin/components/EventDetailsPage/TicketTypesSection/index.tsx:26]: @pauloiankoski when I click "Create ticket" it redirects to the Events List Table with the Create Event modal open.

Comment (kjohnson) [reply src/EventTickets/resources/admin/components/EventDetailsPage/TicketTypesSection/index.tsx:26]: Disregard.

**Review by kjohnson — APPROVED** [review]

**Pull request merged**